cartographer_ros/cartographer_ros/node.cc

```diff
@@ -185,27 +187,31 @@ bool Node::HandleSubmapQuery(
     ::cartographer_ros_msgs::SubmapQuery::Request& request,
     ::cartographer_ros_msgs::SubmapQuery::Response& response) {
   carto::common::MutexLocker lock(&mutex_);
-  return map_builder_bridge_->HandleSubmapQuery(request, response);
+  return map_builder_bridge_.HandleSubmapQuery(request, response);
 }
 
 bool Node::HandleFinishTrajectory(
     ::cartographer_ros_msgs::FinishTrajectory::Request& request,
-    ::cartographer_ros_msgs::FinishTrajectory::Response& response) {
+    ::cartographer_ros_msgs::FinishTrajectory::Response&) {
   carto::common::MutexLocker lock(&mutex_);
-  return map_builder_bridge_->HandleFinishTrajectory(request, response);
+  const int previous_trajectory_id = trajectory_id_;
+  trajectory_id_ = map_builder_bridge_.AddTrajectory(
+      expected_sensor_ids_, options_.tracking_frame, &tf_buffer_);
```

Comment:
Since the bridge is the only user of the `tf_buffer_`: Move it there?
Otherwise, since it never changes: Pass to the constructor?